src/v/pandaproxy/auth_utils.h

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+#include "config/rest_authn_endpoint.h"
+#include "pandaproxy/types.h"
+#include "redpanda/request_auth.h"
+
+#include <seastar/http/request.hh>
+
+#include <vector>
+
+namespace pandaproxy {
+namespace {
```

Comment:
You could make this `inline` or split the definition into a a `.cc` file.

https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rs-unnamed

Reply:
agree. make the method inline (or move the implementation into a cc file). i'm not sure there is any value in having the anonymous namespace in a header. at least i think about it as a replacement for static scoping in a translation unit, but not in a header.